src/pam_google_authenticator.c

```diff
@@ -1561,6 +1561,8 @@ static int google_authenticator(pam_handle_t *pamh, int flags,
 
   // Read and process status file, then ask the user for the verification code.
   int early_updated = 0, updated = 0;
+
+  long hotp_counter = 0;
   if ((username = get_user_name(pamh, &params)) &&
```

Comment:
Rate limit exceeded should not allow an OTP attempt. Should be possible to just move the rate limit before the `secret=`, I think.